src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -164,20 +171,87 @@ struct
                          applying the diff does not match blockchain state's \
                          ledger hash and staged ledger hash resp.\n")))
           in
-          {transition_with_hash; staged_ledger= transitioned_staged_ledger} )
+          { transition_with_hash
+          ; staged_ledger= transitioned_staged_ledger
+          ; ledger_proof_txns } )
 
     let hash {transition_with_hash; _} = With_hash.hash transition_with_hash
 
     let parent_hash {transition_with_hash; _} =
       Consensus.Protocol_state.previous_state_hash
-        (Inputs.External_transition.Verified.protocol_state
-           (With_hash.data transition_with_hash))
+        ( With_hash.data transition_with_hash
+        |> Inputs.External_transition.Verified.protocol_state )
+
+    (** Given:
+     *
+     *       bads
+     *        o       o
+     *       /       /
+     *    o ---- o -
+     *  root \  heir \
+     *        o       o
+     *               heir_heirs
+     *
+     *  commit heir into root and reparent heir_heirs onto root throw away bads
+    *)
+    let commit ~root ~heir ~bads ~heir_heirs :
+        t * t list * Transaction.t Non_empty_list.t option =
+      (* We shouldn't be merging breadcrumbs if the earlier one emitted a proof
+       * it needs to be committed into the snarked ledger *)
+      assert (Option.is_some root.ledger_proof_txns) ;
+      let f = Fn.compose Inputs.Staged_ledger.ledger staged_ledger in
```

Comment:
Can this not just be named `f`? It's convenient and short, but this code would read more clearly if this was even just called `ledger` or `staged_ledger` for instance.